src/net_processing.cpp

```diff
@@ -3579,6 +3719,18 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         pfrom.fSuccessfullyConnected = true;
+
+        if (pfrom.IsSensitiveRelayConn()) {
+            // The peer may intend to later send us NetMsgType::FEEFILTER limiting
+            // cheap transactions, but we don't wait for that and thus we may send
+            // them a transaction below their threshold. This is ok because this
+            // relay logic is designed to work even in cases when the peer drops
+            // the transaction (due to it being too cheap, or for other reasons).
+            PushUnbroadcastTx(pfrom, *peer, interruptMsgProc);
+
+            MaybeSendPing(pfrom, *peer, GetTime<std::chrono::microseconds>());
```

Comment:
Would it make any sense here to manually set `peer.m_ping_queued = true` to guarantee (not "maybe") sending ping?

Reply:
Right! Better to ensure explicitly than rely on the internals of `MaybeSendPing()`, the clock and the fact this is the first call to `MaybeSendPing()` for this peer.